.github/workflows/ci.yml

```diff
@@ -0,0 +1,228 @@
+
+# Abstract:
+#  - This CI is executed when a new commit is created on the main branch or
+#    on a PR whose head branch is the main branch.
+#  - However, the CI will not be executed if files not directly related to
+#    source code maintenance are updated.
+
+name: Files CI
+
+on:
+  push:
+    branches:
+      - main
+    paths-ignore:
+      - 'specs/**'
+      - '*.md'
+  pull_request:
+    paths-ignore:
+      - 'specs/**'
+      - '*.md'
+
+run-name: ${{ github.event_name == 'pull_request' && 'Files PR Validation' || 'Files CI Validation' }}
+
+env:
+  WORKING_DIR: ${{ github.workspace }} # Default: 'D:\a\Files\Files'
+  SOLUTION_PATH: '${{ github.workspace }}\Files.sln'
+  PACKAGE_PROJECT_DIR: '${{ github.workspace }}\src\Files.App (Package)'
+  PACKAGE_PROJECT_PATH: '${{ github.workspace }}\src\Files.App (Package)\Files.Package.wapproj'
+  AUTOMATED_TESTS_ARCHITECTURE: 'x64'
+  AUTOMATED_TESTS_CONFIGURATION: 'Release'
+  AUTOMATED_TESTS_PROJECT_DIR: '${{ github.workspace }}\tests\Files.InteractionTests'
+  AUTOMATED_TESTS_PROJECT_PATH: '${{ github.workspace }}\tests\Files.InteractionTests\Files.InteractionTests.csproj'
+  AUTOMATED_TESTS_ASSEMBLY_DIR: '${{ github.workspace }}\artifacts\TestsAssembly'
+  ARTIFACTS_STAGING_DIR:  '${{ github.workspace }}\artifacts'
+  APPX_PACKAGE_DIR: '${{ github.workspace }}\artifacts\AppxPackages'
+  APPX_SELFSIGNED_CERT_PATH: '${{ github.workspace }}\builds\Files_SelfSigned.pfx'
+  WINAPPDRIVER_EXE86_PATH: 'C:\Program Files (x86)\Windows Application Driver\WinAppDriver.exe'
+  WINAPPDRIVER_EXE64_PATH: 'C:\Program Files\Windows Application Driver\WinAppDriver.exe'
+
+jobs:
+
+  check-formatting:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: pwsh
+
+    steps:
+
+    - name: Checkout the repository
+      uses: actions/checkout@v3
+    
+    - name: Install XamlStyler console
+      run: 'dotnet tool install --global XamlStyler.Console'
+
+    - name: Check XAML formatting
+      id: check-step
+      run: |
+        $changedFiles = (git diff --name-only HEAD~1) -split "\n" | Where-Object {$_ -like "*.xaml"}
+        foreach ($file in $changedFiles)
+        {
+          xstyler -p -l None -f $file
+          if ($LASTEXITCODE -ne 0)
+          {
+            echo "::error file=$file::Format check failed"
+          }
+        }
+      continue-on-error: true
+    
+    - name: Fail the job
+      if: steps.check-step.outcome == 'failure'
+      run: exit 1
+
+  build:
+    runs-on: windows-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        configuration: [Debug, Release]
+        platform: [x64, arm64]
+    env:
+      CONFIGURATION: ${{ matrix.configuration }}
+      ARCHITECTURE: ${{ matrix.platform }}
+
+    steps:
+
+    - name: Checkout the repository
+      uses: actions/checkout@v3
+
+    - name: Setup MSBuild
+      uses: microsoft/setup-msbuild@v1
+
+    - name: Setup NuGet
+      uses: NuGet/setup-nuget@v1.1.1
+
+    - name: Setup .NET 7
```

Comment:
Setup .NET should be moved above installing XamlStyler

Reply:
But installing .NET is not check-xaml-formatting.yml
Do you know why this have kept failing even thought I did completely copied the same to here?

Reply:
It's because the action checks out with a depth of 1, but the formatting part needs the last two commits to format only the XAML files changed in the last commit
I did it this way so it's faster, but it may be worth changing it to do the entire repo if it doesn't have a significant impact

Reply:
Thank you! I disagree with make it run entirely because we already must have ran this checking every time we merge a PR and I believe we don't need to ensure again every time someone create a commmit within their PR.